### PR TITLE
Cleanup x/warden

### PR DESCRIPTION
**Pull request by pitasi**
I bundled together a few cleanups and minor things affecting the x/warden module. See the list of commits to see what changed.

## Summary by CodeRabbit

- **New Features**
  - Updated terminology for clarity, renaming "parties" to "writers" across the application.
  - Introduced explicit state transitions for `KeyRequestStatus` and `SignRequestStatus`.

- **Improvements**
  - Enhanced fee flexibility by changing `Keychain.Fees` type to `Coins`.
  - Improved clarity and consistency in comments and log messages.
  - Refined fields in `KeyRequest` and `Key` messages for better state management.

- **Bug Fixes**
  - Removed unused fields `Keychain.IsActive` and `Keychain.AdminIntentId`.

- **Refactor**
  - Updated various functions and methods to reflect new terminology and field changes.

### CODE REVIEW

File: warden/x/warden/keeper/intents.go [hunk 1]
```diff
@@ -69,15 +69,6 @@ func (k Keeper) newSignatureRequestIntent(ctx context.Context, msg *v1beta2.MsgN
 		return nil, intenttypes.Intent{}, err
 	}
 
-	keychain, err := k.keychains.Get(ctx, key.KeychainId)
-	if err != nil {
-		return nil, intenttypes.Intent{}, err
```
Comment (artur-abliazimov) [inline warden/x/warden/keeper/intents.go:74]: Do we need to at least check that keychain exists?

**Comment by artur-abliazimov** [comment]
I think we do not need the explicit conversion here:

https://github.com/warden-protocol/wardenprotocol/blob/698a5c137ece9d2b63dd3b57c8b2da5a4c6ff006/warden/x/warden/types/v1beta2/keychain.go#L13

Looks like this code is not used anywhere:

https://github.com/warden-protocol/wardenprotocol/blob/698a5c137ece9d2b63dd3b57c8b2da5a4c6ff006/warden/x/warden/types/v1beta2/keys.go#L11

https://github.com/warden-protocol/wardenprotocol/blob/698a5c137ece9d2b63dd3b57c8b2da5a4c6ff006/warden/x/warden/types/v1beta2/keys.go#L18

https://github.com/warden-protocol/wardenprotocol/blob/698a5c137ece9d2b63dd3b57c8b2da5a4c6ff006/warden/x/warden/types/v1beta2/params.go#L10

https://github.com/warden-protocol/wardenprotocol/blob/698a5c137ece9d2b63dd3b57c8b2da5a4c6ff006/warden/x/warden/types/v1beta2/errors.go#L12

File: warden/x/warden/keeper/intents.go (hunk 1 above)
Comment (pitasi) [reply warden/x/warden/keeper/intents.go:74]: if the Key exists it means it came from an existing Keychain

**Comment by pitasi** [comment]
> I think we do not need the explicit conversion here:
> 
> https://github.com/warden-protocol/wardenprotocol/blob/698a5c137ece9d2b63dd3b57c8b2da5a4c6ff006/warden/x/warden/types/v1beta2/keychain.go#L13
> 
> Looks like this code is not used anywhere:
> 
> https://github.com/warden-protocol/wardenprotocol/blob/698a5c137ece9d2b63dd3b57c8b2da5a4c6ff006/warden/x/warden/types/v1beta2/keys.go#L11
> 
> https://github.com/warden-protocol/wardenprotocol/blob/698a5c137ece9d2b63dd3b57c8b2da5a4c6ff006/warden/x/warden/types/v1beta2/keys.go#L18
> 
> https://github.com/warden-protocol/wardenprotocol/blob/698a5c137ece9d2b63dd3b57c8b2da5a4c6ff006/warden/x/warden/types/v1beta2/params.go#L10
> 
> https://github.com/warden-protocol/wardenprotocol/blob/698a5c137ece9d2b63dd3b57c8b2da5a4c6ff006/warden/x/warden/types/v1beta2/errors.go#L12

fixed them all, thanks :)

**Review by artur-abliazimov — APPROVED** [review]

**Pull request merged**